### PR TITLE
Added plugin receiver docs folder

### PR DESCRIPTION
**Pull request by asenkowski**
### Proposed Change
Added plugins folder to docs for plugin receiver documentation.

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Review by StefanKurek — APPROVED** [review]

File: docs/plugins/syslog.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+| `listen_ip`        | `0.0.0.0`        | A network interface for the agent to bind. Typically 0.0.0.0 for most configurations.  |
+| `connection_type`  | `udp`            | Transport protocol to use (`udp` or `tcp`)                                             |
+| `protocol`         | `rfc5424 (IETF)` | Protocol of received syslog messages (`rfc3164 (BSD)` or `rfc5424 (IETF)`)             |
+| `location`         | `UTC`            | [Geographic location (timezone)](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) to use when [parsing the timestamp](https://github.com/observIQ/stanza/blob/master/docs/types/timestamp.md) (Syslog RFC 3164 only) |
+| `tls_enable`       | `false`          | Set to `true` to enable TLS |
+| `tls_certificate`  |                  | Path to x509 PEM encoded TLS certificate file |
+| `tls_private_key`  |                  | Path to x509 PEM encoded TLS private key file |
+| `tls_min_version`  | `"1.2"`          | Minimum TLS version to support (string)       |
+
```
Comment (BinaryFissionGames) [inline docs/plugins/syslog.md:18]: We should also document:
- `data_flow` parameter
- `max_log_size` parameter

File: docs/plugins/tomcat.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# `tomcat` plugin
+
+The `tomcat` plugin consumes [Apache Tomcat](https://tomcat.apache.org/) log entries from the local filesystem and outputs parsed entries.
+
+## Configuration Fields
+
+| Field | Default | Description |
+| --- | --- | --- |
+| `source` | `file` | Use this field to specify where your logs are coming from. When choosing the 'file' option, the agent reads in logs from the log paths specified below. When choosing the 'Kubernetes' options, the agent reads logs from /var/log/containers based on the Pod and Container specified below. |
```
Comment (BinaryFissionGames) [inline docs/plugins/tomcat.md:9]: No k8s support
```suggestion
```

File: docs/plugins/tomcat.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+| `log_format` | `default`  | When choosing the 'default' option, the agent will expect and parse logs in a format that matches the default logging configuration. When choosing the 'observIQ' option, the agent will expect and parse logs in an optimized JSON format that adheres to the observIQ specification, requiring an update to the server.xml file. |
```
Comment (BinaryFissionGames) [inline docs/plugins/tomcat.md:10]: Looks like the alternate log format was dropped too
```suggestion
```

File: docs/plugins/tomcat.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+| `enable_access_log` | `true` | Enable to collect Apache Tomcat access logs |
+| `access_log_path` | `"/usr/local/tomcat/logs/localhost_access_log.*.txt"` | Path to access log file |
+| `enable_catalina_log` | `true` | Enable to collect Apache Tomcat catalina logs |
+| `catalina_log_path` | `"/usr/local/tomcat/logs/catalina.out"` | Path to catalina log file |
+| `cluster_name` | `""` | Friendly name to be used as a resource label. Only relevant if the source is "kubernetes". |
+| `pod_name` | `'tomcat-*'` | The pod name (without the unique identifier on the end). Only relevant if the source is "kubernetes". |
+| `container_name` | `"*"` | The container name of the Nginx container. Only relevant if the source is "kubernetes". |
```
Comment (BinaryFissionGames) [inline docs/plugins/tomcat.md:17]:
```suggestion
```

File: docs/plugins/tomcat.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+| `start_at` | `end` | Start reading file from 'beginning' or 'end' |
```
Comment (BinaryFissionGames) [inline docs/plugins/tomcat.md:18]: Also has a timezone param
```suggestion
| `start_at` | `end` | Start reading file from 'beginning' or 'end' |
| `timezone` | `timezone` | `UTC` | Timezone to use when parsing the timestamp |
```

**Review by BinaryFissionGames — CHANGES_REQUESTED** [review]
Tried to be thorough on all the differences I can see here between these docs and the actual plugins (although I'm sure I missed some); It turns out there are quite a few, which makes sense, since they are all based off the stanza docs.

File: docs/plugins/active_directory.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# `active_directory` plugin
```
[the remark this answers is not in the document]
Comment (cpheps) [reply docs/plugins/active_directory.md:1]: I agree with @BinaryFissionGames. If we do want a single file we should split it into a logs and metrics section. Possibly others if we have specific use case plugins down the road.

File: docs/plugins/active_directory.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+## Configuration Fields
+
+| Field | Type | Default | Description |
+| --- | --- | --- | --- |
+| `enable_dns_server` | `bool` | `true` | Enable to collect DNS server logs |
+| `enable_dfs_replication` | `bool` | `true` | Enable to collect DFS replication logs |
+| `enable_file_replication` | `bool` | `false` | Enable to collect file replication logs |
+| `poll_interval` | `string` | `1s` | Set the rate that logs are being collected  |
+| `max_reads` | `int` | `1000` | Maximum number of logs collected |
+| `start_at` | `enum` | `end` | Start reading file from 'beginning' or 'end' |
```
[the remark this answers is not in the document]
Comment (cpheps) [reply docs/plugins/active_directory.md:18]: In the [parameter types](https://github.com/observIQ/observiq-otel-collector/tree/main/receiver/pluginreceiver#parameters) for the plugin receiver docs we don't call out enums. We also don't have great documentation on writing your own plugin but that's a separate issue.

Just saying `string` for the type and listing possible values should be good. As long as it's hard for the user to make a mistake.

**Comment by cpheps** [comment]
Closing this as we're currently working on a way to autogenerate documentation.

**Pull request closed without merging**